api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -50,7 +51,14 @@ Table createTable(
    * @param tableIdentifier an identifier to identify this table in a namespace.
    * @return true if table exists, false if it doesn't.
    */
-  boolean tableExists(TableIdentifier tableIdentifier);
+  default boolean tableExists(TableIdentifier tableIdentifier) {
+    try {
```

Comment:
The implementation here simplifies code but also introduces exception-handling to ordinary control flow. An alternative is to leave the method blank and give the implementors discretion about how to do it.

Reply:
I think it is reasonable to do this in a default implementation. You can't expect the default implementation to do everything the best way.